data-azure-cosmos/src/main/java/io/micronaut/data/cosmos/operations/DefaultCosmosRepositoryOperations.java

```diff
@@ -223,17 +234,22 @@ public <R> Page<R> findPage(PagedQuery<R> query) {
     public <T> T persist(InsertOperation<T> operation) {
         CosmosContainer container = getContainer(operation);
         T entity = operation.getEntity();
+        RuntimePersistentEntity persistentEntity = runtimeEntityRegistry.getEntity(entity.getClass());
         ObjectNode tree = serializeToTree(entity, Argument.of(operation.getRootEntity()));
-        container.createItem(tree, PartitionKey.NONE, new CosmosItemRequestOptions());
+        PartitionKey partitionKey = PartitionKey.NONE;
```

Comment:
Similar to findByOne here, if PartitionKey is defined on a container and there is no way to supply it in this method then we need to extract it manually from the bean.
If there is a way to pass PartitionKey please let me know.